### DISCUSSION TITLE
Best way to blank an x-modelled inputs value when x-show is false

### DISCUSSION THREAD

**Discussion by npw1982** [alpinejs/alpine#3421] [1. Help]
Hi gang, I've been using alpine for a new system we are developing, and it's all been very helpful so far, but I've gotten stuck trying to work out the best way to achieve the following without lots of manual effort on the Devs part.

We have lots of long forms (and most of what we do is long complex forms). x-show is nice and readable and inline for the display rules, so all good there. 

However when a control is hidden, we need to set it's value to blank "". We have been doing this in the on change event in the parent control, but there is so much repetition that it is becoming nigh on unmaintainable. Plus we have grandparent controls that also need to blank data in the grandchild controls, which leads to even more repetition.

What would be ideal is to have a separate script that dynamically identifies and blanks all the controls with x-show == false, or with display: none. That way we can trigger it for all the controls in one go and remove a whole swathe of repetition. I don't know if anyone with more experience of Alpine has any ideas of the best way to achieve this? Help!

**Comment by ekwoka** [comment]
The best way is to not use a different value for x-show.

Just have the x-show be based o there being a value.

Alternatively, you can use `Alpine.effect` or `x-effect`

**Reply by npw1982** [reply]
Thanks for the reply, x-effect looks like it might be useful.

Apologies I don't think I explained the scenario very well, x-show shouldn't be based on there being a value as the visibility is based on a separate x-model, as below (we are using the Aire form builder).

So in the below case `participant_death` has been mapped with x-bind by Aire automatically. The x-show sections on the child controls are fine, but I would like to removed the need for the x-change element on the parent if possible.

````php
    {{ Aire::select(__('global.option.yesNo'), 'participant_death', __('changestatus.participant_death'))
        ->id('participant_death')
        ->prependEmptyOption('Please Select', '')
        ->setAttribute('x-on:change', "if(!(participant_death == 1)){
            death_date = '';
            death_cause = '';
        }")
        ->defaultValue('') }}
    {{ Aire::date('death_date', __('changestatus.death_date'))
            ->id('death_date')
            ->groupSetAttribute('x-show', 'participant_death == 1') }}
    {{ Aire::input('death_cause', __('changestatus.death_cause'))
            ->id('death_cause')
            ->type('text')
            ->groupSetAttribute('x-show', 'participant_death == 1') }}
````
